.github/workflows/publish-image.yml

```diff
@@ -47,7 +47,7 @@ jobs:
 
       - name: Build and push Docker image
         uses: docker/build-push-action@3b5e8027fcad23fda98b2e3ac259d8d67585f671
-        if: github.ref_type == 'tag'
+        if: github.ref_type == 'tag' || github.ref == 'refs/heads/main'
```

Comment:
You probably also want to modify the `meta` action above to give the container some sensible tags and labels?